FirebaseMLModelDownloader/Sources/ModelInfoRetriever.swift

```diff
@@ -281,45 +281,107 @@ class ModelInfoRetriever {
                                     message: ModelInfoRetriever.DebugDescription
                                       .modelInfoUnmodified,
                                     messageCode: .modelInfoUnmodified)
+              self.telemetryLogger?.logModelInfoRetrievalEvent(eventName: .modelDownload,
+                                                               status: .modelInfoRetrievalSucceeded,
+                                                               errorCode: .noError)
               completion(.success(.notModified))
             case 400:
-              let description = ModelInfoRetriever.ErrorDescription.invalidModelName(self.modelName)
-              DeviceLogger.logEvent(level: .debug,
-                                    message: description,
-                                    messageCode: .invalidModelName)
+              if let data = data,
+                let responseJSON = try? JSONSerialization
```

Comment:
nit: looks like the response parsing and logging is repeated several times, so worth refactoring to a separate method/helper/closure to reduce duplication.

Reply:
Done! I have made this change but make it part of a future PR and not push here so as not to disturb merging the dependent PR #7464.